### PR TITLE
refactor: remove `ref` and `deref` expressions

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@sockmaster27 Do you think the failures here are just some timeouts? Because the tests that failed are unrelated to my changes.

File: main/src/ca/uwaterloo/flix/language/phase/EffectVerifier.scala [hunk 1]
```diff
@@ -225,20 +225,6 @@ object EffectVerifier {
       expectType(expected, actual, loc)
     case Expr.VectorLength(exp, loc) =>
       visitExp(exp)
-    case Expr.Ref(exp1, exp2, tpe, eff, loc) =>
-      visitExp(exp1)
-      visitExp(exp2)
-      // TODO region stuff
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/EffectVerifier.scala:231]: I like when my old TODOs magically disappear

File: main/src/ca/uwaterloo/flix/language/phase/Parser2.scala [hunk 2]
```diff
@@ -1581,8 +1581,7 @@ object Parser2 {
              | TokenKind.TripleTilde
              | TokenKind.KeywordLazy
              | TokenKind.KeywordForce
-             | TokenKind.KeywordDiscard
-             | TokenKind.KeywordDeref => unaryExpr()
+             | TokenKind.KeywordDiscard => unaryExpr()
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Parser2.scala:1584]: Weird that `ref` and `deref` are not treated the same.

File: main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala [hunk 3]
```diff
@@ -1574,7 +1574,7 @@ class TestRedundancy extends AnyFunSuite with TestUtils {
         |def hof(f: a -> b \ e, x: a): b \ e = f(x)
         |
         |def f(): Unit =
-        |    hof(x -> (x, ref 21 @ Static));
+        |    hof(x -> (x, Ref.fresh(21, Static));
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala:1577]: Wrong order. Should be `Static, 21`

File: main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala [hunk 4]
```diff
@@ -565,14 +565,14 @@ class TestTyper extends AnyFunSuite with TestUtils {
         |pub def f(): Int32 =
         |    let _ = {
         |        region rc {
-        |            let x = ref 123 @ rc;
+        |            let x = Ref.fresh(123, rc);
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala:568]: idem

File: main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala [hunk 5]
```diff
@@ -582,14 +582,14 @@ class TestTyper extends AnyFunSuite with TestUtils {
...
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala:585]: idem

File: main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala [hunk 6]
```diff
@@ -599,14 +599,14 @@ class TestTyper extends AnyFunSuite with TestUtils {
...
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala:602]: idem

File: main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala [hunk 7]
```diff
@@ -616,17 +616,17 @@ class TestTyper extends AnyFunSuite with TestUtils {
...
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala:619]: idem

File: main/src/ca/uwaterloo/flix/language/phase/Parser2.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser2.scala:1584]: Yeah. Not sure. But it does not matter now.

File: main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala (hunk 3 above)
Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala:1577]: Good catch! Fixed

**Review by mlutze — APPROVED** [review]

**Pull request merged**